tests/integration/delta_lake/test_table_read.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import annotations
+
+import datetime
+
+import pandas as pd
+import pytest
+
+deltalake = pytest.importorskip("deltalake")
+
+import daft
+from daft.logical.schema import Schema
+
+
+@pytest.fixture(params=[1, 2, 10])
+def local_deltalake_table(request, tmp_path) -> deltalake.DeltaTable:
+    path = tmp_path / "some_table"
+    base_df = pd.DataFrame(
+        {
+            "a": [1, 2, 3],
+            "b": [1.0, 2.0, 3.0],
+            "c": ["foo", "bar", "baz"],
+            "d": [True, False, True],
+            "e": [datetime.datetime(2024, 2, 10), datetime.datetime(2024, 2, 11), datetime.datetime(2024, 2, 12)],
+        }
+    )
+    dfs = []
+    for part_idx in range(request.param):
+        part_df = base_df.copy()
+        part_df["part_idx"] = pd.Series([part_idx] * 3)
+        deltalake.write_deltalake(path, part_df, mode="append")
```

Comment:
you should add some partition columns otherwise the predicate pushdown wont really do much

Reply:
got it